ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13600,7 +13616,12 @@ ASTNode analyzeCreateTable(
 
     // Handle different types of CREATE TABLE command
     // Note: each branch must call addDbAndTabToOutputs after finalizing table properties.
-
+    Database database  = getDatabase(qualifiedTabName.getDb());
+    boolean isDefaultTableTypeChanged = false;
+    if(database.getParameters() != null && database.getParameters().containsKey(DEFAULT_TABLE_TYPE) && database.getParameters().get(DEFAULT_TABLE_TYPE).equalsIgnoreCase("external")) {
+      isExt = true;
+      isDefaultTableTypeChanged = true;
```

Comment:
appears the logic when "defaultTableType=ACID" is not implemented. I do not see where we check for this and convert to ACID automatically.

Reply:
Ack. Addressed the comment in next commit